### PR TITLE
Iss742: Add condition for SVT sync status

### LINKED ISSUE #742: Add condition for SVT sync status
table: svt_sync_statuses
field: good

If good is true then the corrections do not need to be applied. 

False indicates corrections are needed.

----

List of good and bad runs here:

https://confluence.slac.stanford.edu/pages/viewpage.action?pageId=295094627

### PR DESCRIPTION
**Pull request by JeremyMcCormick**
The SVT sync status condition is represented by the class `SvtSyncStatus` with the method `isGood()` returning `true` if the status is good and `false` if not.

Data is contained in the table `svt_sync_statuses` in the primary JLAB conditions database and was loaded from the information provided by @normangraf (see comments on the github issue).

There is a test that performs a spot check on a few of the records, which can be run as follows:

```
mvn test -Dtest=SvtSyncStatusesTest
```

This successfully checks that the first and last conditions records are correct (good and bad, respectively).

### CODE REVIEW

**Comment by normangraf** [comment]
RawTrackerHitFitterDriver.java fails the style checker and does not compile. Please fix.

File: tracking/src/main/java/org/hps/recon/tracking/RawTrackerHitFitterDriver.java
Comment (mholtrop) [inline tracking/src/main/java/org/hps/recon/tracking/RawTrackerHitFitterDriver.java]: This looks like it changes the tt offset by 2 from what was done previously. I would expect at least some comment why this new behavior is superior than what was done before.

Comment (cbravo135) [reply tracking/src/main/java/org/hps/recon/tracking/RawTrackerHitFitterDriver.java]: This was discussed in multiple meetings. This shifts the average offset to be roughly zero, rather than the -2 it used to be. The intention was always to have a average time of roughly zero, and not -2.

Comment (cbravo135) [reply tracking/src/main/java/org/hps/recon/tracking/RawTrackerHitFitterDriver.java]: I would like to see a comment as to why 12 was chosen in the first place, and why it wasn't immediately fixed when it was seen that the behavior of the code was not what was actually intended.

**Comment by normangraf** [comment]
All of the 2015 & 2016 integration tests fail. This should not happen, as this issue was intended to resolve problems specific to the 2019 data.

File: tracking/src/main/java/org/hps/recon/tracking/RawTrackerHitFitterDriver.java
Comment (cbravo135) [reply tracking/src/main/java/org/hps/recon/tracking/RawTrackerHitFitterDriver.java]: When I brought up it being at -2, it was argued it didn't matter because it changes the times the same amount and we only use relative hit times, not the absolute times, in our selection. I would say the same argument applies to this scenario, and zero is better because then defining a cut on the absolute time becomes a more simple cut.

**Comment by cbravo135** [comment]
@normangraf this is an inaccurate representation of the intention of this issue.

**Comment by cbravo135** [comment]
The intention was to improve the calibration of the hit time corrections.

**Comment by JeremyMcCormick** [comment]
Can you paste the log from the test failures here?

**Comment by cbravo135** [comment]
https://srs.slac.stanford.edu/hudson/view/HPS/job/hps-java-dev/62/testReport/

**Comment by normangraf** [comment]
Failed tests:
  EngRun2015FeeReconTest.testIt:67->comparePlots:106 expected:<1> but was:<103>
  EngRun2015MollerReconTest.testIt:62->comparePlots:101 expected:<417> but was:<389>
  EngRun2015V0ReconTest.testIt:66->comparePlots:105 expected:<1949> but was:<1957>
  PhysRun2016FeeReconTest.testIt:62->comparePlots:89 expected:<125> but was:<128>
  PhysRun2016MollerReconTest.testIt:66->comparePlots:93 expected:<2221> but was:<2182>
  PhysRun2016V0ReconTest.testIt:66->comparePlots:93 expected:<4890> but was:<4743>

**Comment by JeremyMcCormick** [comment]
This must be a consequence of changing the raw tracker hit fitting. Just
adding the sync condition would not do this.

**Comment by normangraf** [comment]
That's correct. Just adding the SVT conditions should have no effect.

**Comment by cbravo135** [comment]
If you look in jenkins, you can see that if I change the 14 back to 12, it passes everything.

**Comment by cbravo135** [comment]
This is the same code, just with the 14 changed to a 12: https://srs.slac.stanford.edu/hudson/job/hps-java-dev/61/

**Review by normangraf — DISMISSED** [review]
Please revert the offset to 12, after which we can approve this pull request and merge with master. If needed, we can open a new issue to deal with the question of whether the offset should be 12 or 14.

**Comment by cbravo135** [comment]
We already did discuss changing this to 14 and the analysis group agreed the intention of this number is to put the average hit time at zero. The offset should be 14, and always should have been 14.

**Comment by cbravo135** [comment]
Mathematically, it makes sense for it to be 14 since the trigger fw and code is in essence "rounding up" the time to the next trigger clock cycle, so it will be off by half of a 4 ns cycle (2 ns) plus the time to the center (from the later edge) of the svt trigger window which is 24 ns wide, so 2 ns plus 12 ns is 14 ns.

**Comment by cbravo135** [comment]
12 is a bug, and I don't want to change my code to have a bug over not having one.

**Comment by cbravo135** [comment]
Seems like it is the tests that should change, not the 14.

**Comment by mholtrop** [comment]
I agree, the test should change. However, we cannot merge this until that is done. Can the test change be part of the pull request, or is that separate?

**Comment by cbravo135** [comment]
The tests can change as part of this PR. I just don't know how to do it myself, so someone that understands how the test framework works would need to do it.

File: conditions/src/main/java/org/hps/conditions/svt/SvtSyncStatus.java
Comment (cbravo135) [inline conditions/src/main/java/org/hps/conditions/svt/SvtSyncStatus.java]: This throws an exception: 
```java.lang.ClassCastException: Cannot cast java.lang.Integer to java.lang.Boolean```

**Comment by cbravo135** [comment]
So, after reading the code for the Recon tests that are failing, we just need to update some of the reference files in: http://www.lcsim.org/test/hps-java/referencePlots/
Not sure who has perms to do this, but I have no idea how to upload anything there.
It does seem that there is no need to change any code to get the tests to pass, so we can safely merge this PR.

**Comment by cbravo135** [comment]
The tests are now changed and legacy drivers were added for 2015/2016 recon.

**Review by cbravo135 — APPROVED** [review]

**Comment by JeremyMcCormick** [comment]
My original PR added only a small bit of functionality to the conditions system. The relevant branch has now been used to add code that was far beyond the scope of what I originally intended, so I'm closing this. (Also, @cbravo135 please don't approve PRs where you are the primary author of most of the changes, even if they have been originally opened by someone else,)

**Comment by cbravo135** [comment]
@JeremyMcCormick I was asked to review this PR by the software coordinator. Please don't close PRs without talking to the people involved first.

**Comment by JeremyMcCormick** [comment]
@cbravo135 If you want this branch merged then open a PR using your account, not mine. DO NOT REOPEN THIS AGAIN.

**Comment by JeremyMcCormick** [comment]
@cbravo135 You reviewing this PR is nonsense since you are the author of most of the changes.

**Comment by cbravo135** [comment]
The fact of the matter is this PR is still small. Please tell that to @normangraf who directly asked me to provide a review.

**Review by mholtrop — APPROVED** [review]

**Pull request merged**
